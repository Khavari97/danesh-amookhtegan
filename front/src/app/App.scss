@import 'assets/fonts/fonts';

* {
  margin: 0;
  font-family: "Shabnam", serif;
}

a {
  text-decoration: none;
}

//Slider css
.alice-carousel__dots {
  .alice-carousel__dots-item:not(.__custom).__active {
    background-color: #046B00;
  }
}


