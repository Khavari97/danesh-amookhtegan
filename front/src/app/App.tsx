import React from "react";
import "./App.scss";
import AppRouter from "./navigation/AppRouter";
import {SnackbarProvider} from 'notistack';
import {SnackbarUtils} from "./components/material-ui/Snackbar/SnackbarUtils";
import ThemeProvider from "./assets/styles/material/material";

export interface AppProps {
}

export interface AppState {
}

class App extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);
        this.state = {};
    }

    componentDidMount = () => {
    }

    render() {
        return (
            <SnackbarProvider maxSnack={3} anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}>
                <SnackbarUtils/>
                <ThemeProvider>
                    <AppRouter/>
                </ThemeProvider>
            </SnackbarProvider>
        );
    }
}

export default App;
