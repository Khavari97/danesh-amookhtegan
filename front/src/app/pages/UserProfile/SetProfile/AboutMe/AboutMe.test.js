import React from 'react';
import Enzyme, { shallow, configure, mount } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { Provider } from "react-redux";
import App from "./AboutMe";
import renderer from "react-test-renderer";
import { render, screen, act } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { Button } from '@material-ui/core';
import thunk from 'redux-thunk'
import configureStore from 'redux-mock-store';
configure({ adapter: new Adapter() })

const middlewares = [thunk]
const mockStore = configureStore(middlewares);
const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-testid="${val}"]`);
};



describe('test set profile states', () => {
    let wrapper
    let store_mock
    const data = {
        description: "دانشجو"
    }

    store_mock = mockStore({ userProfile: { loading: false, description: data.description } })
    store_mock.dispatch = jest.fn()

    wrapper = mount(
        <BrowserRouter>
            <Provider store={store_mock}>
                <App />
            </Provider>
        </BrowserRouter>
    );


    test('check show props', () => {
        expect(findByTestAttr(wrapper, 'desc').text()).toEqual(data.description)
    })
    test('snapshot', () => {
        const snapshot = renderer.create(wrapper).toJSON();
        expect(snapshot).toMatchSnapshot();
    })

});