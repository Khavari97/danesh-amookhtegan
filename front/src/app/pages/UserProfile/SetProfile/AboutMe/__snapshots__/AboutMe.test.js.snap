// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test set profile states snapshot 1`] = `
<div
  className="MuiGrid-root animate__animated animate__fadeIn MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
  >
    <div
      className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="title"
          data-testid="desc"
        >
          دانشجو
        </p>
      </li>
    </div>
  </div>
</div>
`;
